core/src/appleMain/kotlin/CentralManager.kt

```diff
@@ -6,25 +6,48 @@ import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.withContext
 import platform.CoreBluetooth.CBCentralManager
+import platform.CoreBluetooth.CBCentralManagerOptionRestoreIdentifierKey
 import platform.CoreBluetooth.CBCharacteristic
 import platform.CoreBluetooth.CBCharacteristicWriteType
 import platform.CoreBluetooth.CBDescriptor
 import platform.CoreBluetooth.CBPeripheral
 import platform.CoreBluetooth.CBService
 import platform.CoreBluetooth.CBUUID
 import platform.Foundation.NSData
+import platform.Foundation.NSUUID
+import platform.Foundation.NSUserDefaults
 
 private const val DISPATCH_QUEUE_LABEL = "central"
+private const val CBCENTRALMANAGER_RESTORATION_ID = "kable-central-manager"
+private const val CBCENTRALMANAGER_CONSUMER_ID_KEY = "kable-central-manager-consumer-id"
 
 public class CentralManager internal constructor() {
 
     internal companion object {
         val Default: CentralManager by lazy { CentralManager() }
     }
 
+    private val userDefaults = NSUserDefaults.standardUserDefaults
+
+    // This value is needed to ensure multiple instances of Kable running on the same iOS device
+    // do not cross pollinate restored instances of CBCentralManager. The value will live for the
+    // lifetime of the consuming app.
+    private val consumerId: String
+        get() {
+            val id = userDefaults.stringForKey(CBCENTRALMANAGER_CONSUMER_ID_KEY)
+            return if (id != null) {
+                id
+            } else {
+                val id = NSUUID().UUIDString()
+                userDefaults.setObject(id, CBCENTRALMANAGER_CONSUMER_ID_KEY)
+                id
+            }
```

Comment:
Nit: intent is a little clearer with mutation

```suggestion
            var id = userDefaults.stringForKey(CBCENTRALMANAGER_CONSUMER_ID_KEY)
            if (id == null) {
                id = NSUUID().UUIDString()
                userDefaults.setObject(id, CBCENTRALMANAGER_CONSUMER_ID_KEY)
            }
            return id
```

Reply:
Also nit: even clearer in point-free style IMHO:
```suggestion
            get() = userDefaults.stringForKey(CBCENTRALMANAGER_CONSUMER_ID_KEY)
                ?: NSUUID().UUIDString().also {
                    userDefaults.setObject(it, CBCENTRALMANAGER_CONSUMER_ID_KEY)
                }
```